les/odr_requests.go

```diff
@@ -487,7 +487,7 @@ func (r *TxStatusRequest) GetCost(peer *serverPeer) uint64 {
 
 // CanSend tells if a certain peer is suitable for serving the given request
 func (r *TxStatusRequest) CanSend(peer *serverPeer) bool {
-	return peer.serveTxLookup
```

Comment:
Tx serving can be entirely disabled (for example in case of "announce only" ULC servers, though I'm not sure if anyone uses that feature at the moment). You should return `peer.txHistory != txIndexDisabled` here.